common/lib/capa/capa/responsetypes.py

```diff
@@ -1010,7 +1114,16 @@ def setup_response(self):
             contextualize_text(choice.get('name'), self.context)
```

Comment:
This function is now complex enough it really needs a doc string. It took me a bit of time to understand it, and some of the nuances (e.g. the default=0.5 only applies if partial is set on a choice). A simple explanation of `correct_choices`, `partial_choices`, and `partial_values` would be nice.